core/templates/dev/head/components/share/SharingLinksDirective.js

```diff
@@ -22,24 +22,48 @@ oppia.directive('sharingLinks', [function() {
     scope: {
       layoutType: '@',
       layoutAlignType: '@',
+      shareType: '@',
       getTwitterText: '&twitterText',
-      getExplorationId: '&explorationId'
+      getExplorationId: '&explorationId',
+      getCollectionId: '&collectionId'
     },
     templateUrl: 'components/sharingLinks',
     controller: [
-      '$scope', '$window', 'oppiaHtmlEscaper', 'ExplorationEmbedButtonService',
+      '$scope', '$window', '$injector', 'oppiaHtmlEscaper',
       'siteAnalyticsService', 'UrlInterpolationService',
       function(
-          $scope, $window, oppiaHtmlEscaper, ExplorationEmbedButtonService,
+          $scope, $window, $injector, oppiaHtmlEscaper,
           siteAnalyticsService, UrlInterpolationService) {
-        $scope.explorationId = $scope.getExplorationId();
+        $scope.registerShareEvent = null;
 
-        $scope.registerShareExplorationEvent = function(network) {
-          siteAnalyticsService.registerShareExplorationEvent(network);
-        };
+        if ($scope.shareType === 'exploration') {
+          $scope.explorationId = $scope.getExplorationId();
+          var ExplorationEmbedButtonService = $injector.get(
+            'ExplorationEmbedButtonService');
 
-        $scope.showEmbedExplorationModal = (
-          ExplorationEmbedButtonService.showModal);
+          $scope.activityType = 'explore';
+          $scope.activityId = $scope.explorationId;
+
+          $scope.registerShareEvent = function(network) {
```

Comment:
You can just say `$scope.registerShareEvent = siteAnalyticsService.registerShareExplorationEvent`, I think -- functions are first-class objects in JavaScript. (Make sure you test that it still works, though, e.g. by adding a console.log in registerShareExplorationEvent().)

Ditto below.